### PR TITLE
Add FastTree

### LINKED ISSUE #47: add FastTree
It's the default for tree-building, so I would like it.
It's under the GPLv2, as shown on the top of the source code: 
http://www.microbesonline.org/fasttree/FastTree.c 

Links here: http://www.microbesonline.org/fasttree/#Install

### PR DESCRIPTION
**Pull request by colinbrislawn**
Closes #47

### CODE REVIEW

File: recipes/FastTree/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  fn: FastTree-2.1.8.c
+  url: http://www.microbesonline.org/fasttree/FastTree-2.1.8.c
+
+build:
+  number: 0
+  script:
+    - mkdir -p "$PREFIX/bin"                                  # [osx or linux]
+
+    - gcc -DUSE_DOUBLE -O3 -finline-functions -funroll-loops -Wall -o FastTree FastTree-2.1.8.c -lm
+    - gcc -fopenmp -DUSE_DOUBLE -O3 -finline-functions -funroll-loops -Wall -o FastTreeMP FastTree-2.1.8.c -lm
```
Comment (colinbrislawn) [inline recipes/FastTree/meta.yaml]: This line usually starts `-DOPENMP -fopenmp`. I've omitted `-DOPENMP` to make it compile using the default OSX gcc. idk if's true OpenMP or not. 
On Travis-CI's version of OSX, `gcc` is an alias for `clang`, which does not yet support OpenMP.

File: recipes/FastTree/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    - mv -v ./FastTree* $PREFIX/bin                           # [osx or linux]
+    - xcopy /s "FastTree.exe" "%PREFIX%\bin"                  # [win64]
+
+test:
+  commands:
+    - FastTree -help                                          # [osx or linux]
+    - FastTreeMP -help                                        # [osx or linux]
```
Comment (colinbrislawn) [inline recipes/FastTree/meta.yaml]: A real test here would be good. Especially a test for OpenMP and `export OMP_NUM_THREADS=2`.

**Comment by colinbrislawn** [comment]
I think OpenMP is only readably supportable on Linux. Everyone fine if we only build OpenMP on Linux?

**Comment by wasade** [comment]
Nah, should ship by default with osx unless I'm mistaken

**Comment by wasade** [comment]
Huh, maybe not. Just prodded the Google's. That's surprising. Okay, ya, I
guess both MP and non

**Comment by colinbrislawn** [comment]
I tried to find an easy way of spinning up OpenMP on osx using conda. Got
nothing. But as far as I know, most people do heavy bioinformatics on Linux
where OpenMP is common.

**Comment by colinbrislawn** [comment]
Anyone have a good input alignment to use as a test? I'll try using the one in the qiime unit test on `make_phylogeny.py`, but it seems a little small. Maybe that's not a problem. 
https://github.com/biocore/qiime/blob/master/tests/test_make_phylogeny.py#L121

**Comment by wasade** [comment]
Random subsample from Greengenes?

**Comment by colinbrislawn** [comment]
@wasade Not a bad idea. What's a good subsampling size? The FastTree documentation lists examples, but many are very big.

Perhaps a quick unit test, like the one in qiime, is good enough?
How should we test FastTreeMP? (It's non-deterministic.)

**Comment by wasade** [comment]
Meh, just 10 seqs?

Topology should be the same I think with MP

**Comment by colinbrislawn** [comment]
Tree-building weirdness:
The version of fasttree that ships with qiime and the newest version I build here produce different results. The qiime version is 2.1.3 which did not support double precision. 

```
+ diff aln_for_tree.out aln_for_tree.tre
1c1
< (ghi:0.000000005,def:0.000000005,(jkl:0.402823848,abc:0.072531513)0.734:0.071522450);
---
> (ghi:0.00014,def:0.00014,(jkl:0.40282,abc:0.07253)0.742:0.07152);
```

Should I build the default qiime version or a newer, better one?

**Comment by wasade** [comment]
Use newer. Flagging @gregcaporaso re: qiime shipped version not using DP

**Comment by colinbrislawn** [comment]
haha, oh I went with FastTree version 2.1.3 for consistency with my qiime install.

Is my qiime install using an older version of FastTree. Double precision was added in 2.1.8...

**Comment by wasade** [comment]
Okay, I highly suggest we use the DP version...

**Comment by colinbrislawn** [comment]
Here is where qiime-1.9.1 downloads FastTree 2.1.3
https://github.com/biocore/qiime/blob/c2b62a86100f08e26011a11587a8da60ccaf409a/setup.py#L209 

I think moving to DP and better options are a great (see biocore/qiime#466). Let's build the current version first so that it's on Binstar, then build the newer, better version.

**Comment by wasade** [comment]
Ugh, that sucks, qiime hard codes 2.1.3.

**Comment by colinbrislawn** [comment]
I've squashed my commits. Ready to merge! :shipit:

**Comment by colinbrislawn** [comment]
Once merged, I'm ready to push FastTree 2.1.8 with unit tests and compiled with double precision. :shipit:

**Comment by RNAer** [comment]
looks awesome. thanks, @colinbrislawn ! merging

**Pull request merged**